### PR TITLE
feat(server): support AWS Cognito auth server

### PR DESCRIPTION
**Pull request by yk-eukarya**

### CODE REVIEW

File: server/internal/app/config/mailer.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+	Name  string
+}
+
+type SMTPConfig struct {
+	Host         string
+	Port         string
+	SMTPUsername string
+	Email        string
+	Password     string
+}
```
Comment (rot1024) [inline server/internal/app/config/mailer.go:20]: They are not used. `reearthx/mailer` should be used. You can delete them.

File: server/internal/app/config.go [hunk 2]
```diff
@@ -49,163 +44,21 @@ type Config struct {
...
-	AuthSrv       AuthSrvConfig
+	Auth          config.AuthConfigs
+	Auth0         config.Auth0Config
+	Cognito       config.CognitoConfig
+	AuthSrv       config.AuthSrvConfig
 	Auth_ISS      string
 	Auth_AUD      string
 	Auth_ALG      *string
 	Auth_TTL      *int
 	Auth_ClientID *string
```
Comment (rot1024) [inline server/internal/app/config.go:60]: This Config struct can also be transfered to `config` package

File: server/internal/app/config.go [hunk 3]
```diff
@@ -261,21 +114,41 @@ func (c Config) Print() string {
...
+		u = nil
+	}
+	return u
+}
+
+func (c Config) Auths() (res []config.AuthConfig) {
 	if ac := c.Auth0.AuthConfigs(); len(ac) > 0 {
 		res = append(res, ac...)
 	}
+	if ac := c.Cognito.AuthConfig(); ac != nil {
```
Comment (rot1024) [inline server/internal/app/config.go:137]:
```go
type AuthProvider interface {
	Config() []AuthConfig
}
```

can be defined in the `config` package to simplify Auths:

```go
func (c Config) AuthConfigs() []AuthProvider {
	return []AuthProvider{c.Auth0, c.Cognito}
}

func (c Config) Auths() (res []config.AuthConfig) {
	ac := lo.FlatMap(c.AuthConfigs(), func(c AuthConfig) []AuthConfig { return c.Config() }
	// ...
}
```

**Review by rot1024 — APPROVED** [review]
Creating the config package is a very good idea.

File: server/internal/app/config/mailer.go (hunk 1 above)
Comment (yk-eukarya) [reply server/internal/app/config/mailer.go:20]: Done

**Pull request merged**